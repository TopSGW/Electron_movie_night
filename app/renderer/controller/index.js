// Logic for mapping application state onto component props.
import {
  App,
  DisplayMoviesContainer,
  ImportMovies,
  ImportStats,
  SearchMovies,
  MainContent,
  MovieThumbnailContainer
} from './containers'

// Logic for mapping application events directly to redux dispatch actions.
import './mapEventsToDispatch'

// Re-export all the containers.
export {
  App,
  DisplayMoviesContainer,
  ImportMovies,
  ImportStats,
  MainContent,
  MovieThumbnailContainer,
  SearchMovies
}
