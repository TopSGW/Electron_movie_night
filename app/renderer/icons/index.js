import cn from 'classnames'
import React from 'react'

const Icon = ({ className, border, fixedWidth, large, pulse, size3x, ...props }) => {
  return <i
    className={cn(
      'fa',
      className,
      pulse && 'fa-pulse',
      large && 'fa-lg',
      fixedWidth && 'fa-fw',
      size3x && 'fa-3x',
      border && 'fa-border'
    )}
    {...props}
  />
}

export const Anchor = ({ className, ...props }) => {
  return <Icon className={cn('fa-anchor', className)} {...props} />
}

export const Angle = ({ className, down, left, right, up, ...props }) => {
  return <Icon
    className={cn(
      down && 'fa-angle-down',
      left && 'fa-angle-left',
      right && 'fa-angle-right',
      up && 'fa-angle-up',
      !down && !left && !right && !up && 'fa-angle-right', // default
      className
    )}
    {...props}
  />
}

export const Bars = ({ className, ...props }) => {
  return <Icon className={cn('fa-bars', className)} {...props} />
}

export const Ban = ({ className, ...props }) => {
  return <Icon className={cn('fa-ban', className)} {...props} />
}

export const Chevron = ({ className, down, left, right, up, ...props }) => {
  return <Icon
    className={cn(
      down && 'fa-chevron-down',
      left && 'fa-chevron-left',
      right && 'fa-chevron-right',
      up && 'fa-chevron-up',
      !down && !left && !right && !up && 'fa-chevron-right', // default
      className
    )}
    {...props}
  />
}

export const Database = ({ className, ...props }) => {
  return <Icon className={cn('fa-database', className)} {...props} />
}

export const Close = ({ className, ...props }) => {
  return <Icon className={cn('fa-close', className)} {...props} />
}

export const Edit = ({ className, ...props }) => {
  return <Icon className={cn('fa-pencil', className)} {...props} />
}

export const FileIcon = ({ className, video, ...props }) => {
  return <Icon
    className={cn(
      video && 'fa-file-video-o',
      !video && 'fa-file', // default
      className
    )}
    {...props}
  />
}

export const ImportIcon = ({ className, ...props }) => {
  return <Icon
    className={cn(
      'fa-download',
      className
    )}
    {...props}
  />
}

export const Info = ({ className, circle, ...props }) => {
  return <Icon
    className={cn(
      circle && 'fa-info-circle',
      !circle && 'fa-info',
      className
    )}
    {...props}
  />
}

export const Play = ({ className, ...props }) => {
  return <Icon className={cn('fa-play', className)} {...props} />
}

export const Search = ({ className, ...props }) => {
  return <Icon className={cn('fa-search', className)} {...props} />
}

export const Spinner = ({ className, ...props }) => {
  return <Icon className={cn('fa-spinner', className)} {...props} />
}

export const Star = ({ className, half, outline, ...props }) => {
  return <Icon
    className={cn(
      half && 'fa-star-half-o',
      outline && 'fa-star-o',
      !half && !outline && 'fa-star', // default
      className
    )}
    {...props}
  />
}

export const Trash = ({ className, ...props }) => {
  return <Icon className={cn('fa-trash', className)} {...props} />
}
