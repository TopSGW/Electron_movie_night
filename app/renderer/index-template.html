<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <title>Movie Night</title>
    <link rel="stylesheet" href="../vendor/font-awesome-4.7.0/css/font-awesome.min.css">
    <%=htmlWebpackPlugin.options.css%>
  </head>
  <body style="background-color:rgb(20,20,20) !important; /* override bootstrap */">
    <div id="root"></div>
    <script src="<%=htmlWebpackPlugin.options.bundle%>"></script>
  </body>
</html>
