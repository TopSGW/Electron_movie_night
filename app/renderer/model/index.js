import { ipcRenderer } from 'electron'
import { createStore, applyMiddleware, combineReducers } from 'redux'
import reduxLogger from 'redux-logger'

import { MOVE_MOVIE_TO_TRASH, UPDATE_MOVIE_METADATA } from '../../shared/events'
import { isDevEnv } from '../../shared/utils'
import appLogger from '../mainWindowLogger'

// Reducers
import dbLoaded from './dbLoaded'
import crawl from './crawl'
import featuredMovie from './featuredMovie'
import movies from './movies'
import search from './search'

// Top level state selectors.
import {
  crawlActiveSelector,
  crawlStatsSelector,
  showCrawlStatsSelector,
  movieDBSelector,
  searchCategorySelector,
  searchQuerySelector
} from './selectors'

// Memoized Selectors
import enhancedFeaturedMovieSelector from './enhancedFeaturedMovie'
import visibleMoviesSelector from './visibleMovies'

// Re-export all the action creators.
// ----------------------------------
export * from './actions'

// Electron action creators.
// -------------------------
export const updateMetadataFor = (movie) => {
  ipcRenderer.send(UPDATE_MOVIE_METADATA, movie)
  appLogger.info('Sent UPDATE_MOVIE_METADATA event', { movie: movie.title })
}

export const deleteMovie = (movie) => {
  ipcRenderer.send(MOVE_MOVIE_TO_TRASH, movie)
  appLogger.info(`Sent ${MOVE_MOVIE_TO_TRASH} event`, { movie: movie.title })
}

// Public selectors.
// These allow connected containers to depend on state
// without requiring knowledge of the state's shape.
// ----------------------------------------------------------------------
export const getAllMovies = (state) => {
  return movieDBSelector(state)
}

export const getCrawlActive = (state) => {
  return crawlActiveSelector(state)
}

export const getCrawlStats = (state) => {
  return crawlStatsSelector(state)
}

export const getShowCrawlStats = (state) => {
  return showCrawlStatsSelector(state)
}

export const getFeaturedMovie = (state) => {
  return enhancedFeaturedMovieSelector(state)
}

export const getSearchCategory = (state) => {
  return searchCategorySelector(state)
}

export const getSearchQuery = (state) => {
  return searchQuerySelector(state)
}

export const getVisibleMovies = (state) => {
  return visibleMoviesSelector(state)
}

// Export function for creating redux store.
// -----------------------------------------
export const createReduxStore = () => {
  // Create the redux store.
  // This is the default export.
  // ---------------------------
  let middlewares = []
  if (isDevEnv()) {
    middlewares.push(reduxLogger)
  }

  const rootReducer = combineReducers({
    crawl,
    dbLoaded,
    featuredMovie,
    movies,
    search
  })

  const store = createStore(rootReducer, applyMiddleware(...middlewares))
  return store
}

// This is the actual store used in the app.
// -----------------------------------------
const store = createReduxStore()
export default store
