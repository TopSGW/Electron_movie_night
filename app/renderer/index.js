import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'

import { logEnv } from '../shared/utils'
import { App } from './controller'
import store from './model'
import logger from './mainWindowLogger'

logEnv(logger)

// Render the App.
ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
)
