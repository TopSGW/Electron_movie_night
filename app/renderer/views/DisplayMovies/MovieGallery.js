import React, { Component } from 'react'
import styled from 'styled-components'
import { Chevron } from '../../icons'
import { MovieThumbnailContainer } from '../../controller'

import { fadeIn } from '../styleUtils'

export default class MovieGallery extends Component {
  constructor (props) {
    super(props)
    this.selectCategory = this.selectCategory.bind(this)
  }

  selectCategory (category) {
    return (e) => {
      e.preventDefault()
      const { handleSelectCategory } = this.props
      if (handleSelectCategory) {
        handleSelectCategory(category)
      }
    }
  }

  render () {
    const { category } = this.props

    return (
      <Gallery key={category}>
        { this.renderTitle() }
        { this.renderThumbnails() }
      </Gallery>
    )
  }

  renderTitle () {
    const { category, renderStyle } = this.props
    if (!category) {
      return null
    }

    if (renderStyle === 'grid') {
      return <Title>{category}</Title>
    }

    return (
      <Title>
        <Genre onClick={this.selectCategory(category)}>
          {category}
          <Chevron right fixedWidth />
        </Genre>
      </Title>
    )
  }

  renderThumbnails () {
    const { movies, renderStyle, id } = this.props

    const thumbnails = movies.map((movie) => {
      return (
        <FadeIn key={movie.location}>
          <MovieThumbnailContainer
            // key={movie.location}
            movie={movie}
            panelID={id}
          />
        </FadeIn>
      )
    })

    if (renderStyle === 'grid') {
      return <GridContainer>{thumbnails}</GridContainer>
    } else {
      return <HorizontalScrollContainer>{thumbnails}</HorizontalScrollContainer>
    }
  }
}

const FadeIn = styled.div`
  animation: 0.5s ${fadeIn} ease-in;
`

const Gallery = styled.div`
  margin-top: 30px;
  margin-right: 10px;
`

const Title = styled.h2`
  line-height: 1.3;
  margin: 0;
  display: block;
  font-size: 1.2em;
  font-weight: bold;
  color: #999;
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  background-color: transparent;
  text-transform: capitalize;
`

const Genre = styled.span`
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  &:hover {
    color: rgba(255, 255, 255, 0.7);
  }
`

const HorizontalScrollContainer = styled.div`
  display: flex;
  overflow-x: auto;
  &::-webkit-scrollbar {
    height: 12px;
  }

  &::-webkit-scrollbar-button {
    display: none;
  }

  &::-webkit-scrollbar-track {
    display: none;
  }

  &::-webkit-scrollbar-thumb {
    display: block;
    border-radius: 10px;
    background-color: rgba(255, 255, 255, 0.1);
  }
`

const GridContainer = styled.div`
  display: flex;
`
