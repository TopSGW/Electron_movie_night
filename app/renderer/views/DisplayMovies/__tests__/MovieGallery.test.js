'use strict'
/* globals jest, describe, test, expect */

import React from 'react'
import renderer from 'react-test-renderer'

import MovieGallery from '../MovieGallery'

jest.mock('../../../controller', () => {
  return {
    MovieThumbnailContainer: ({ movie, panelID }) => {
      return <div>MovieThumbnailContainerMock</div>
    }
  }
})

const movie = {
  actors: ['Ben Stiller', 'Will Ferrell'],
  director: 'Todd Phillips',
  imgFile: '/path/to/img',
  fileInfo: [
    { location: '/path/to/movie' }
  ],
  genres: ['action', 'comedy', 'drama'],
  imdbID: 'tt123'
}

describe('MovieGallery', () => {
  test('render horizontal scroll with no category', () => {
    const component = renderer.create(
      <MovieGallery
        movies={[movie]}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('render grid style with category', () => {
    const component = renderer.create(
      <MovieGallery
        movies={[movie]}
        category={'Drama'}
        renderStyle={'grid'}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('render category plus callback selectCategory onClick', () => {
    const selectCategory = jest.fn()

    const component = renderer.create(
      <MovieGallery
        movies={[movie]}
        category={'Action'}
        handleSelectCategory={selectCategory}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    const title = tree.children[0]
    const chevron = title.children[0]
    chevron.props.onClick({ preventDefault: jest.fn() })
    expect(selectCategory).toHaveBeenLastCalledWith('Action')
  })
})
