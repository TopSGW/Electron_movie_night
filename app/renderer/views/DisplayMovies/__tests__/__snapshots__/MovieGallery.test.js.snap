// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGallery render category plus callback selectCategory onClick 1`] = `
<div
  className="sc-bwzfXH jOuPFw"
>
  <h2
    className="sc-htpNat gvHaIo"
  >
    <span
      className="sc-bxivhb bSYJga"
      onClick={[Function]}
    >
      Action
      <i
        className="fa fa-chevron-right fa-fw"
      />
    </span>
  </h2>
  <div
    className="sc-ifAKCX kxzXyO"
  >
    <div
      className="sc-bdVaJa erRPjQ"
    >
      <div>
        MovieThumbnailContainerMock
      </div>
    </div>
  </div>
</div>
`;

exports[`MovieGallery render grid style with category 1`] = `
<div
  className="sc-bwzfXH jOuPFw"
>
  <h2
    className="sc-htpNat gvHaIo"
  >
    Drama
  </h2>
  <div
    className="sc-EHOje fhxHOS"
  >
    <div
      className="sc-bdVaJa erRPjQ"
    >
      <div>
        MovieThumbnailContainerMock
      </div>
    </div>
  </div>
</div>
`;

exports[`MovieGallery render horizontal scroll with no category 1`] = `
<div
  className="sc-bwzfXH jOuPFw"
>
  <div
    className="sc-ifAKCX kxzXyO"
  >
    <div
      className="sc-bdVaJa erRPjQ"
    >
      <div>
        MovieThumbnailContainerMock
      </div>
    </div>
  </div>
</div>
`;
